_search-plugins/knn/nested-search-knn.md

```diff
@@ -0,0 +1,347 @@
+        "_id": "2",
+        "_score": 0.0040983604,
+        "_source": {
+          "nested_field": [
+            {
+              "my_vector": [
+                10,
+                10,
+                10
+              ]
+            },
+            {
+              "my_vector": [
+                20,
+                20,
+                20
+              ]
+            },
+            {
+              "my_vector": [
+                30,
+                30,
+                30
+              ]
+            }
+          ]
+        }
+      }
+    ]
+  }
+}
+```
+
+## k-NN search with filtering on nested fields
+
+You can apply a filter to k-NN search with nested fields. A filter can be applied to either a top-level field or a field inside a nested field.
+
+The following example applies a filter to a top-level field. 
+
+First, create a k-NN index with a nested field:
+
+```json
+PUT my-knn-index-1
+{
+  "settings": {
+    "index": {
+      "knn": true
+    }
+  },
+  "mappings": {
+    "properties": {
+      "nested_field": {
+        "type": "nested",
+        "properties": {
+          "my_vector": {
+            "type": "knn_vector",
+            "dimension": 3,
+            "method": {
+              "name": "hnsw",
+              "space_type": "l2",
+              "engine": "lucene",
+              "parameters": {
+                "ef_construction": 100,
+                "m": 16
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+After you create the index, add some data to it:
+
+```json
+PUT _bulk?refresh=true
+{ "index": { "_index": "my-knn-index-1", "_id": "1" } }
+{"parking": false, "nested_field":[{"my_vector":[1,1,1]},{"my_vector":[2,2,2]},{"my_vector":[3,3,3]}]}
+{ "index": { "_index": "my-knn-index-1", "_id": "2" } }
+{"parking": true, "nested_field":[{"my_vector":[10,10,10]},{"my_vector":[20,20,20]},{"my_vector":[30,30,30]}]}
+{ "index": { "_index": "my-knn-index-1", "_id": "3" } }
+{"parking": true, "nested_field":[{"my_vector":[100,100,100]},{"my_vector":[200,200,200]},{"my_vector":[300,300,300]}]}
+```
+{% include copy-curl.html %}
+
+Then run a k-nearest neighbors search on the data using the `knn` query type with a filter. The following query returns documents whose `parking` field is set to `true`:
+
+```json
+GET my-knn-index-1/_search
+{
+  "query": {
+    "nested": {
+      "path": "nested_field",
+      "query": {
+        "knn": {
+          "nested_field.my_vector": {
+            "vector": [
+              1,
+              1,
+              1
+            ],
+            "k": 3,
+            "filter": {
+              "term": {
+                "parking": true
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Even though all three nearest vectors to the query vector are in document 1, the query returns documents 2 and 3 because document 1 is filtered out:
```

Comment:
```suggestion
Even though all three vectors nearest to the query vector are in document 1, the query returns documents 2 and 3 because document 1 is filtered out:
```